### PR TITLE
fix: Make sure the service is up to date every time

### LINKED ISSUE #4287: When the deployment selector is updated, dapr will get a "connection refused" error
dapr suddenly appeared inaccessible, check the debug logs found that it is due to the deployment of the selector property change, dapr service is not updated caused

> /area operator

## What version of Dapr?

 version 1.6.0 -- commit 4bb25fab444c4f1a1bf0ffd74293dbd4fdcea580"

## Expected Behavior

Service should be updated

## Actual Behavior

https://github.com/dapr/dapr/blob/master/pkg/operator/handlers/dapr_handler.go#L160-L167

Service will only be created based on the app-id and will not be updated later

## Steps to Reproduce the Problem

- Deploy a deployment with dapr enabled
- Update the deployment selector matchLabels
- Update the deployment

### LINKED ISSUE #4304: When version exists in labels of Deployment, dapr may not be able to complete the creation of svc
/area operator

> 1.6.0

brief  deploy yml 
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: test
    version: v3-1-1236
  name: test-3.1.1236
  namespace: dev
spec:
  selector:
    matchLabels:
      app: test
      version: v3-1-1236
  template:
    metadata:
      annotations:
        dapr.io/app-id: test
        dapr.io/app-port: "80"
        dapr.io/enabled: "true"
        dapr.io/http-max-request-size: "1000"
        dapr.io/log-level: debug
      labels:
        app: test
        version: v3-1-1236
    spec:
      containers:
      - image: test:v3-1-1236
```
svc yml at this time
```
#brief
apiVersion: v1
kind: Service
metadata:
  annotations:
    dapr.io/app-id:test
    prometheus.io/path: /
    prometheus.io/port: "9090"
    prometheus.io/scrape: "true"
  labels:
    dapr.io/enabled: "true"
  name: test-dapr
  namespace: dev
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: Deployment
    name: test-3.1.1236
```

1.If I add a new version `Deployment`(3-1-1237)
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: test
    version: v3-1-1237
  name: test-3.1.1237
  namespace: dev
spec:
  selector:
    matchLabels:
      app: test
      version: v3-1-1237
  template:
    metadata:
      annotations:
        dapr.io/app-id: test
        dapr.io/app-port: "80"
        dapr.io/enabled: "true"
        dapr.io/http-max-request-size: "1000"
        dapr.io/log-level: debug
      labels:
        app: test
        version: v3-1-1237
    spec:
      containers:
      - image: test:v3-1-1237
```
2. Then delete the previous version `Deployment`(3.1.1236)
3. `test-dapr` will not be created at this time

### PR DESCRIPTION
**Pull request by alberthuang24**
# Description

Keep the service up to date every time. prevent the service from remaining old when the deployment is updated

## Issue reference

fix #4287
fix #4304

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/operator/handlers/dapr_handler.go [hunk 1]
```diff
@@ -165,6 +165,21 @@ func (h *DaprHandler) ensureDaprServicePresent(ctx context.Context, namespace st
 		log.Errorf("unable to get service, %s, err: %s", mayDaprService, err)
 		return err
 	}
+
+	h.patchDaprService(ctx, mayDaprService, wrapper)
```
Comment (yaron2) [inline pkg/operator/handlers/dapr_handler.go:169]: Can you check for the error that's returned here?

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/operator/handlers/dapr_handler.go [hunk 2]
```diff
@@ -165,6 +165,24 @@ func (h *DaprHandler) ensureDaprServicePresent(ctx context.Context, namespace st
...
+
+	return nil
+}
+
+func (h *DaprHandler) patchDaprService(ctx context.Context, expectedService types.NamespacedName, wrapper ObjectWrapper) error {
+	appID := h.getAppID(wrapper)
+	service := h.createDaprServiceValues(ctx, expectedService, wrapper, appID)
+
+	if err := h.Update(ctx, service); err != nil {
+		log.Errorf("unable to update service, %s, err: %s", service.Name, err)
```
Comment (CrazyHZM) [inline pkg/operator/handlers/dapr_handler.go:182]: The log is printed repeatedly, it is recommended to remove it.

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

**Comment by CrazyHZM** [comment]
@alberthuang24 
Please fix the DCO.

**Comment by alberthuang24** [comment]
@yaron2 Is there anything else I need to do to move forward with this pr merge?

**Review by yaron2 — APPROVED** [review]

**Comment by yaron2** [comment]
Thanks for your contribution @alberthuang24

**Pull request merged**